application/src/main/java/org/togetherjava/tjbot/commands/basic/RoleSelectCommand.java

```diff
@@ -0,0 +1,262 @@
+            event.getChannel()
+                .sendMessageEmbeds(makeEmbed(title, description))
+                .setActionRow(menu.build())
+                .queue();
+
+            event.reply("Message sent successfully!").setEphemeral(true).queue();
+        }
+    }
+
+    /**
+     * Adds role options to a selection menu
+     * <p>
+     *
+     * @param event the {@link SlashCommandEvent}
+     * @param menu the menu to add options to {@link SelectionMenu.Builder}
+     * @param placeHolder the placeholder for the menu {@link String}
+     * @param minValues the minimum number of selections. nullable {@link Integer}
+     */
+    private static void addMenuOptions(@NotNull SlashCommandEvent event,
+            @NotNull SelectionMenu.Builder menu, @NotNull String placeHolder,
+            @Nullable Integer minValues) {
+
+        Role highestBotRole =
+                Objects.requireNonNull(event.getGuild()).getSelfMember().getRoles().get(1);
+        List<Role> guildRoles = Objects.requireNonNull(event.getGuild()).getRoles();
+
+        List<Role> roles = new ArrayList<>(
+                guildRoles.subList(guildRoles.indexOf(highestBotRole) + 1, guildRoles.size()));
+
+        if (minValues != null) {
+            menu.setMinValues(minValues);
+        }
+
+        menu.setPlaceholder(placeHolder).setMaxValues(roles.size());
+
+        for (Role role : roles) {
+            if (role.getName().equals("@everyone") || role.getTags().getBotId() != null) {
+                continue;
+            }
+            menu.addOption(role.getName(), role.getId());
+        }
+    }
+
+    /**
+     * Creates an embedded message to send with the selection menu
+     *
+     * <p>
+     * </p>
+     *
+     * @param title for the embedded message. nullable {@link String}
+     * @param description for the embedded message. nullable {@link String}
+     * @return the formatted embed {@link MessageEmbed}
+     */
+    private static @NotNull MessageEmbed makeEmbed(@Nullable String title,
+            @Nullable String description) {
+
+        if (title == null) {
+            title = "Select your roles:";
+        }
+
+        return new EmbedBuilder().setTitle(title)
+            .setDescription(description)
+            .setColor(new Color(24, 221, 136))
+            .build();
+    }
+
+    @Override
+    public void onSelectionMenu(@NotNull SelectionMenuEvent event, @NotNull List<String> args) {
+        Member member = Objects.requireNonNull(event.getMember(), "Member is null");
+
+        Guild guild = Objects.requireNonNull(event.getGuild());
+
+        List<Role> selectedRoles = new ArrayList<>();
+        for (SelectOption selectedOption : Objects.requireNonNull(event.getSelectedOptions())) {
+            Role selectedRole = guild.getRoleById(selectedOption.getValue());
+            if (selectedRole != null && guild.getSelfMember().canInteract(selectedRole)) {
+                selectedRoles.add(selectedRole);
+            }
+        }
```

Comment:
Thoughts on using the Stream API instead?

```java
event.getSelectedOptions()
   .stream()
   .map(selectedOption -> guild.getRoleById(selectedOption.getValue())
   .filter(Objects::requireNonNull) 
   .filter(role -> guild.getSelfMember().canInteract(role))
   .toList();
``` 

Not sure how advanced you are with Java, so if you don't understand how this works, feel free to ask ^^